examples/rp_pico/fiber/main.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022, Andrey Kunitsyn
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+#include <modm/debug/logger.hpp>
+#include <modm/processing.hpp>
+#include <modm/platform/core/multicore.hpp>
+#include <mutex>
+
+using namespace Board;
+using namespace std::chrono_literals;
+
+// Create an IODeviceWrapper around the Uart Peripheral we want to use
+modm::IODeviceWrapper< Uart0, modm::IOBuffer::BlockIfFull > loggerDevice;
+
+// Set all four logger streams to use the UART
+modm::log::Logger modm::log::debug(loggerDevice);
+modm::log::Logger modm::log::info(loggerDevice);
+modm::log::Logger modm::log::warning(loggerDevice);
+modm::log::Logger modm::log::error(loggerDevice);
+
+constexpr uint32_t cycles = 1'000'000;
+
+static modm::platform::multicore::Mutex log_mutex;
+
+struct CoreData {
+	uint32_t total_counter = 0;
+	uint32_t f1counter = 0;
+	uint32_t f2counter = 0;
+};
+
+
+void
+fiber_function1(CoreData& d)
+{
+	while (++d.f1counter < cycles) { modm::fiber::yield(); d.total_counter++; }
+}
+
+void
+fiber_function2(CoreData& d)
+{
+	while (++d.f2counter < cycles) { modm::fiber::yield(); d.total_counter++; }
+}
+
+// put cores to mostly equalent environment
+modm_section(".scratch_y_data") static CoreData d0;
+modm_section(".scratch_x_data") static CoreData d1;
```

Comment:
Doesn't x come before y? So it should be CPU0 = x, CPU1 = y?
I really don't like the name scratch_xy, it's non-descriptive.
Maybe `.cpu0_data`, `.cpu1_data` etc? 
(We should still keep the `.scratch_x` sections of course for Pico-SDK compatibility!)

Reply:
`y` for core0 and `x` for core1 from [pico-sdk](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_standard_link/memmap_default.ld#L225), I think it is for use largest address for single core usage.

Reply:
Maybe `.core0_data`, `.core1_data` ?